### ISSUE TITLE
[BUG] Pages with views but no visitors

### ISSUE THREAD

**Issue by stoic-hugo** [benvinegar/counterscale#105]
Is it counting bots? I am seeing hundreds of views with no visitors https://imgur.com/a/OC5iKht. Already, in less than 5 days, those are 531 pageviews with no visitors (and I did not screenshot all of them)

**Comment by benvinegar** [comment]
Sorry, it's hard to parse your screenshot. I see a lot of zeros, but the way this is chopped up (?) I can't tell what I'm looking at.

The code for how it determines visits is here: https://github.com/benvinegar/counterscale/blob/main/app/analytics/collect.ts#L5

It's based on HTTP headers. It presumes that a browser client is passing `If-Modified-Since` properly. If a client is always sending a "fresh" `If-Modified-Since` header, it is basically saying, "I already got this resource recently, do you have a new version?" And the server interprets this to mean that this is a returning user, and does not mark it as a visit. (This is how Counterscale tracks visits without using cookies.)

My guess is that if you're not seeing any visits, this is not a regular browser user, and probably a bot.

Perhaps there's more that can be done to harden this endpoint against bots.

**Comment by ka259** [comment]
> Sorry, it's hard to parse your screenshot. I see a lot of zeros, but the way this is chopped up (?) I can't tell what I'm looking at.

I merged several small screenshots together. Left is visitor, right is views. Hope that helps. So, 0 visitors are recorded, but a number of views are (as high as 80).

> My guess is that if you're not seeing any visits, this is not a regular browser user, and probably a bot (that appears to be aggressively checking for new content).

Interesting view...

> Perhaps there's more that can be done to harden this endpoint against bots.

I am not a dev, so I will leave this to you and the more knowledgeable ones.

**Comment by benvinegar** [comment]
I suspect it's also related to using CSP (#101).

> If it helps, the site being tracked is also a static site on Cloudflare Pages.

Could you share with me the URL of this site? It would let me inspect the outbound requests and see what's going on.

If you'd prefer not to share the URL publicly, you can also email me: `github@benv.ca`.

**Comment by ka259** [comment]
> If you'd prefer not to share the URL publicly, you can also email me: github@benv.ca.

I will coz this is a ghost account anyway (I use Gitlab instead).

**Comment by ka259** [comment]
> If you'd prefer not to share the URL publicly, you can also email me: github@benv.ca.

Done! I just created this account to follow up on a few issues anyway.

**Comment by benvinegar** [comment]
@stoic-hugo Is this issue still happening? I did check out your site after getting your email, saw the behavior, couldn't figure it out quickly ... then got sidetracked.

**Comment by benvinegar** [comment]
Thanks for following up. I'm going to close and someone can reopen if it reappears.
